src/Upload/StreamableUploader.php

```diff
@@ -0,0 +1,165 @@
+<?php
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may ob`tain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Upload;
+
+use GuzzleHttp\Psr7\BufferStream;
+use GuzzleHttp\Psr7\Request;
+
+/**
+ * Uploader that is a special case of the ResumableUploader where we can write
+ * the file contents in a streaming manner.
+ */
+class StreamableUploader extends ResumableUploader
```

Comment:
I think it may be possible to tweak `ResumableUploader` so that it can fit the bill for our use case here. Have you investigated that option?